### DISCUSSION TITLE
Response for Java apis is not working

### DISCUSSION THREAD

**Discussion by NAMSKN** [milvus-io/milvus#29144] [Q&A and General discussion]
ShowCollectionsParam param = ShowCollectionsParam.newBuilder().build();
R<ShowCollectionsResponse> response = client.withTimeout(2, TimeUnit.SECONDS).showCollections(param);

The code works till ShowCollectionsParam this line. When we want to get response it give io.netty error and grpc call error

**Comment by yhmo** [comment]
What does the error show?

**Reply by NAMSKN** [reply]
io.grpc.StatusRuntimeException: UNKNOWN
	at io.grpc.stub.ClientCalls.toStatusRuntimeException(ClientCalls.java:271) ~[grpc-stub-1.46.0.jar:1.46.0]
	at io.grpc.stub.ClientCalls.getUnchecked(ClientCalls.java:252) ~[grpc-stub-1.46.0.jar:1.46.0]
	at io.grpc.stub.ClientCalls.blockingUnaryCall(ClientCalls.java:165) ~[grpc-stub-1.46.0.jar:1.46.0]

Caused by: java.lang.UnsupportedOperationException: null
	at io.grpc.netty.AbstractHttp2Headers.isEmpty(AbstractHttp2Headers.java:40) ~[grpc-netty-1.46.0.jar:1.46.0]
	
	
It tells it doesn't support io.netty and grpc calls. I even tried to change the dependency. 

	<dependency>
			<groupId>io.milvus</groupId>
			<artifactId>milvus-sdk-java</artifactId>
			<version>2.3.2</version>
			<exclusions>
				
				<exclusion>
					<groupId>io.netty</groupId>
					<artifactId>netty-codec</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
		<dependency>
			<groupId>io.netty</groupId>
			<artifactId>netty-codec</artifactId>
			<version>4.1.101.Final</version>
		</dependency>

	
Still it gives same error. Please help!!

**Reply by yhmo** [reply]
@NAMSKN 
You can try this workaround: https://github.com/milvus-io/milvus-sdk-java/issues/706#issuecomment-1851954692

```
<dependency>
	<groupId>io.milvus</groupId>
	<artifactId>milvus-sdk-java</artifactId>
	<version>2.3.3</version>
	<exclusions>
		<exclusion>
			<groupId>io.netty</groupId>
			<artifactId>netty-codec</artifactId>
		</exclusion>
		<exclusion>
			<groupId>io.netty</groupId>
			<artifactId>netty-codec-http2</artifactId>
		</exclusion>
	</exclusions>
</dependency>

<dependency>
	<groupId>io.netty</groupId>
	<artifactId>netty-codec</artifactId>
	<version>4.1.101.Final</version>
</dependency>
<dependency>
	<groupId>io.netty</groupId>
	<artifactId>netty-codec-http2</artifactId>
	<version>4.1.100.Final</version>
</dependency>
```

**Comment by yhmo** [comment]
I have no idea why this error occurs. Search by google I got this issue: https://github.com/grpc/grpc-java/issues/10707. Looks like it is a conflict between grpc-java and spring-boot.
